src/features/user/GiftFunds/Details.tsx

```diff
@@ -1,41 +1,77 @@
-import React, { useContext, ReactElement } from 'react';
+import React, { useContext, ReactElement, useState } from 'react';
 import { UserPropsContext } from '../../common/Layout/UserPropsContext';
-import styles from './GiftFunds.module.scss';
 import { useTranslation } from 'next-i18next';
+import { GiftFunds } from '../../common/types/user';
+import { styled } from '@mui/material';
 
 const Details = (): ReactElement | null => {
   const { user } = useContext(UserPropsContext);
   const { t, ready } = useTranslation('giftfunds');
+  const [validGiftFunds, setValidGiftFunds] = useState<GiftFunds[] | null>(
```

Comment:
Would it make sense to move this logic and the mapping logic to the index file (src/features/user/GiftFunds/index.tsx)? That way, this component can be used to show a single GiftFund Details component, and be dependent only on props.

Reply:
Since the component is not being used anywhere else yet, it won't bring much change logically. Though after the changes, the code would look cleaner and it'll bring more uniformity to the code as we've used props in `DashBoardView` children at other places, so I'll make the respective changes.